File: BaekJoon/problems/5_BinarySearch/1072_스파이더_카드놀이.swift
import Foundation

let input = readLine()!
    .split(separator: " ")
    .map { Int($0)! }

var total = input[0]
var win = input[1]
var current = countRecord(total, win) 

func countRecord(_ total: Int, _ win: Int) -> Int { 
    return Int((Float(win) / Float(total)) * 100)
}

var left = 0
var right = 1000000000
var answer = -1

while left <= right { 

    let mid = (left + right) / 2

    let total = total + mid
    let win = win + mid
    let newRecord = countRecord(total, win)

    if newRecord < current + 1 { 
        left = mid + 1
    } else if newRecord > current + 1 { 
        right = mid - 1
    } else { 
        answer = mid
        right = mid - 1
    }
}

print(answer)

File: BaekJoon/problems/5_BinarySearch/1072_스파이더_카드놀이.py
import sys

x, y = map(int, sys.stdin.readline().split())
z = int((y * 100 / x))

answer = -1

l = 0
r = 1000000000

while l <= r:
    
    mid = (l + r) // 2
    nx = x + mid
    ny = y + mid
    nz = int((ny * 100 / nx))

    if nz < z+1:
        l = mid+1
    else:
        r = mid-1
        answer = mid
    
print(answer)